Source/Moq.Analyzers/NoConstructorArgumentsForInterfaceMockAnalyzer.cs

```diff
@@ -28,38 +30,44 @@ private static void Analyze(SyntaxNodeAnalysisContext context)
         var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
 
         // TODO Think how to make this piece more elegant while fast
-        GenericNameSyntax genericName = objectCreation.Type as GenericNameSyntax;
-        if (objectCreation.Type is QualifiedNameSyntax)
+        GenericNameSyntax? genericName = objectCreation.Type as GenericNameSyntax;
+        if (objectCreation.Type is QualifiedNameSyntax qualifiedName)
         {
-            var qualifiedName = objectCreation.Type as QualifiedNameSyntax;
             genericName = qualifiedName.Right as GenericNameSyntax;
         }
 
         if (genericName?.Identifier == null || genericName.TypeArgumentList == null) return;
 
         // Quick and dirty check
-        if (genericName.Identifier.ToFullString() != "Mock") return;
+        if (!string.Equals(genericName.Identifier.ToFullString(), "Mock", StringComparison.Ordinal)) return;
```

Comment:
If we're going to continue to compare against string literals we should DRY this out

Reply:
Yeah, I think another step to make "soon" is to start to pull things into a set of extension methods on the appropriate syntax or semantic element.